geode-core/src/main/java/org/apache/geode/internal/cache/map/RegionMapPut.java

```diff
@@ -146,9 +146,13 @@ protected void serializeNewValueIfNeeded() {
   @Override
   protected void runWhileLockedForCacheModification(Runnable r) {
     cacheModificationLock.lockForCacheModification(getOwner(), getEvent());
+    final boolean locked = getOwner().lockWhenRegionIsInitializing();
```

Comment:
It would be nice if we could share this code by putting it in AbstractRegionMapPut.
Here is how we could do it. Move the runWhileLockedForCacheModification you have in RegionMapCommitPut to AbstractRegionMapPut. Then change this one to just do the old cacheModLock but in the body do this:
super. runWhileLockedForCacheModification(r);